docs/proposals/respect-rbac-for-resource-exclusions.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Respect RBAC for Resource Inclusions/Exclusions
+
+authors:
+- "@gdsoumya"
+- "@alexmt"
+
+sponsors:
+- TBD
+
+reviewers:
+- TBD
+
+approvers:
+- TBD
+
+creation-date: 2023-05-03
+
+---
+
+# Enhancement Idea
+
+This is a proposal to provide the ability to configure argocd controller, to respect the current RBAC permissions 
+when handling resources besides the already existing resource inclusions and exclusions.
+
+## Summary
+
+Argo CD administrator will be able to configure in `argocd-cm`, whether to enable or disable(default) the feature where the controller will 
+only monitor resources that the current service account allows it to read.
+
+## Motivation
+
+Some users restrict the access of the argocd to specific resources using rbac and this feature will enable them to continue 
+using argocd without having to manually configure resource exclusions for all the resources that they don't want argocd to be managing.
+
+## Proposal 
+
+The configuration for this will be present in the `argocd-cm`, we will add new boolean field `resource.respectRBAC` in the
+cm which can be set to `true` to enable this feature, by default the feature is disabled.
+
+For the implementation there are 3 proposals :
+
+1. Modify `gitops-engine` pkg to make a `SelfSubjectAccessReview` request before adding any resource to the watch list, in this approach we are making an extra
+   api server call to check if controller has access to the resource, this does increase the no. of kubeapi calls made but is more accurate.
+2. Modify `gitops-engine` pkg to check for forbidden/unauthorized errors when listing for resources, this is more efficient approach as the
+   no. of kubeapi calls made does not change, but there is a chance of false positives as similar errors can be returned from kubeapi server or env specific proxies in other situations
+3. Combine approaches 1 and 2, in this controller will check the api response for the list call, and if it receives forbidden/unauthorized it will make the `SelfSubjectAccessReview` call.
+   This approach is accurate and at the same time, only makes extra api calls if the list calls fail in the first place.
```

Comment:
This strongly depends on the use-case of the user and what kind of permissions they are willing to give to a specific Argo CD instance. For example, if there's an instance that is only allowed to manage 10 out of 200 APIs, you would still have 180 additional calls to the SelfSubjectAccessReview.

Reply:
Can we maybe make the use of SubjectSelfAccessReview configurable? Then people can have the trade-off they want - on clusters with a low amount of APIs, they can configure the mechanism to the highest precision while on clusters with high amount of APIs, they trade precision against speed.

Reply:
Sure I think we can have another config option that enables the stricter check. Will update the proposal with this

Reply:
Updated.